### PR TITLE
Indicate support for Sphinx parallel-write

### PR DESCRIPTION
**Pull request by BenWaumans**
Added support for sphinx parallel build to avoid the following warnings:
```
WARNING: the mlx.traceability extension does not declare if it is safe for parallel reading, assuming it isn't - please ask the extension author to check and make it explicit
WARNING: doing serial read
```

I'm not sure if there would be any problem with allowing parallel builds in the traceability plugin.

### CODE REVIEW

**Comment by Letme** [comment]
Was this tested on some of the bigger projects? Otherwise, it would be safer to claim we are not...

**Comment by bavovanachte** [comment]
I've been using it with `-j=auto` for a while now for local builds (not in CI because of the warning and because it's mainly single-core runners). I haven't encountered any issues, but I haven't looked at the actual html results.

I've recently marked sphinx-wavedrom as safe for parallel building. This comment could help: https://github.com/bavovanachte/sphinx-wavedrom/issues/39#issuecomment-995884518

**Comment by JasperCraeghs** [comment]
I don't think that parallel read is supported. See https://github.com/sphinx-doc/sphinx/pull/9050/files

**Comment by JasperCraeghs** [comment]
> Based on that even write is not - since we enable changing attributes via callbacks.

I believe that the `traceability_callback_per_item` callback function is called in the event [doctree-read](https://www.sphinx-doc.org/en/master/extdev/appapi.html?highlight=env_before_read_docs#event-doctree-read), which is in the (parallel) read stage. The secondary callback function `traceability_inspect_item` is called in the writing stage but shouldn't/cannot be used to change `env.traceable_collection`. It should only be used to inspect and report warnings. I think that parallel-write may already be supported.

File: mlx/traceability.py [hunk 1]
```diff
@@ -626,4 +626,8 @@ def setup(app):
                                   innernodeclass=nodes.emphasis,
                                   warn_dangling=True))
 
-    return {'version': version}
+    return {
+        'version': version,
+        'parallel_read_safe': True,
```
Comment (Letme) [inline mlx/traceability.py:631]:
```suggestion
        'parallel_read_safe': False,
```

**Review by Letme — CHANGES_REQUESTED** [review]

**Review by Letme — APPROVED** [review]

**Review by JasperCraeghs — APPROVED** [review]

**Pull request merged**